src/nrniv/netpar.cpp

```diff
@@ -1217,6 +1217,16 @@ int nrn_set_timeout(int timeout) {
 }
 
 void BBS::netpar_solve(double tstop) {
+	// temporary check to be eventually replaced by verify_structure()
+	extern int tree_changed, v_structure_change, diam_changed;
+	if (tree_changed || v_structure_change) {
+	  hoc_execerror("NEURON model internal structures are out of date",NULL);
+        }
+	if (diam_changed) {
+	  recalc_diam();
+	}
```

Comment:
IIUC, this is an error for neuron as well

Reply:
Yes. But I was looking into this today and believe that instead of raising an error,  both here and in pc.nrncore_run it should be possible to properly setup the model without calling finitialize and continue safely. (but the onus would be on the user to be satisified that all relevant variable (state) values be set as desired if the structure changes (and finitialize not called). I have just a few questions in my mind about spikes on the queue if in the middle of a simulation, the user
decides to modify the model and switch to simulation with coreneuron. But all this is for another day.